// timeagojs/timeago.js
let setTimeago = {
    curTime: new Date().getTime(),
    curTimeItem: function () {
        return this.getDateItem(this.curTime)
    },
    
    init: function (timeago) {
        let min = 60 * 1000,
            half = 30 * min,
            hour = 2 * half;
        let timeagoItem = this.getDateItem(timeago);
        let str = '';

        if (this.isThisDay(timeago)) {
            let interval = this.curTime - timeago * 1000;
            if (interval <  min){
                str = '刚刚';
            } else if (interval < hour){
                str = Math.floor(interval / min) + '分钟前';
              if (interval >= half && interval <= half + min){
                  str = '半小时前';
              }
            } else if (interval >= hour){
                str = Math.floor(interval / hour) + '小时前';
            }
        } else if ( this.isYesterday(timeago)) {
            str = '昨天'   + timeagoItem.hour + ':' + timeagoItem.min;
        } else if ( this.isBeforeYesterday(timeago)) {
            str = '前天'   + timeagoItem.hour + ':' + timeagoItem.min;
        } else if (this.isThisYear(timeago)){
            str = timeagoItem.month + '月' + timeagoItem.day + '日';
        } else {
            str = timeagoItem.year + '年' + timeagoItem.month + '月' + timeagoItem.day + '日';
        }

        return str;
    },
    getDateItem: function (timestamp) {
        let value = (''+timestamp).length > 10 ? timestamp : timestamp * 1000;
        let date = new Date(value);
        let h = date.getHours(),
            m = date.getMinutes();

        return {
            year: date.getFullYear(),
            month: date.getMonth() + 1,
            day: date.getDate(),
            hour: h > 9 ? h : '0' + h,
            min: m > 9 ? m : '0' + m
        }
    },
    isThisDay: function (timeago) {
    	let day = this.getDateItem(timeago);
    	let curTimeItem = this.curTimeItem();
        return day.year === curTimeItem.year &&
               day.month === curTimeItem.month &&
               day.day === curTimeItem.day;
    },
    isThisMonth: function (timeago) {
    	let day = this.getDateItem(timeago);
    	let curTimeItem = this.curTimeItem();
        return day.year === curTimeItem.year && day.month === curTimeItem.month;
    },
    isThisYear: function (timeago) {
    	let day = this.getDateItem(timeago);
    	let curTimeItem = this.curTimeItem();
        return day.year === curTimeItem.year;
    },
    isYesterday: function (timeago) {
    	let day = this.getDateItem(timeago);
    	let curTimeItem = this.curTimeItem();
        return this.isThisMonth(timeago) && day.day === curTimeItem.day - 1;
    },
    isBeforeYesterday: function (timeago) {
    	let day = this.getDateItem(timeago);
    	let curTimeItem = this.curTimeItem();
        return this.isThisMonth(timeago) && day.day === curTimeItem.day - 2;
    }
}

export default setTimeago
